### DISCUSSION TITLE
Boundary of dia.Element

### DISCUSSION THREAD

**Discussion by CodeSippr** [clientIO/joint#2041] [Q&A]
Is there a way to change the boundary of an element for connection points without manipulating the BBox (I need the BBox to remain unchanged for other functions)? I have an element that consists of multiple rectangles, two of which are invisible. I want thant Links only be able to connect with the visible ones. thanks

**Comment by kumilingus** [comment]
Please provide an example of such a shape (e.g. JSFiddle).

**Reply by CodeSippr** [reply]
https://jsfiddle.net/CodeSippr/c7v8m0uL/1/

**Reply by kumilingus** [reply]
I meant a working example. Something we can run, see what the issue is, and propose a right solution immediately.

---
No, you don't have to change the bounding box.

There are several [connectionPoints](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#connectionPoints.intro) to choose from, plus you can define a custom one. An example of a custom `connectionPoint`  is shown e.g. in [ERD example](https://www.jointjs.com/demos/er-diagrams).

Perhaps the answer could be found here: https://github.com/clientIO/joint/discussions/2028.
